examples/src/walkable-path.ts

```diff
@@ -0,0 +1,151 @@
+/*!
+ * Copyright 2020 Cognite AS
+ */
+
+import * as THREE from 'three';
+import CameraControls from 'camera-controls';
+import * as reveal from '@cognite/reveal';
+import { CogniteClient } from '@cognite/sdk';
+import { toThreeVector3, CadNode } from '@cognite/reveal/threejs';
+import { vec3 } from 'gl-matrix';
+import { SectorModelTransformation } from '@cognite/reveal/models/cad/types';
+
+CameraControls.install({ THREE });
+
+interface TransitPathResponse {
+  items: [
+    {
+      segments: [
+        {
+          path: [
+            {
+              x: number;
+              y: number;
+              z: number;
+            }
+          ];
+        }
+      ];
+    }
+  ];
+}
+
+interface TransitPathRequest {
+  modelId: number;
+  items: [
+    {
+      from: { x: number; y: number; z: number } | { nodeId: number };
+      to: { x: number; y: number; z: number } | { nodeId: number };
+      movingObjectSize: { diameter: number; height: number };
+    }
+  ];
+}
+
+// Should find a way to add this to maybe cognite-sdk-js later?
+async function getTransitPath(client: CogniteClient, transitRequest: TransitPathRequest) {
+  const project = '3ddemo';
+  const url = `https://api.cognitedata.com/api/playground/projects/${project}/3d/pathfinder/transit`;
+  const response = await client.post<TransitPathResponse>(url, {
+    data: transitRequest
+  });
+  return response;
+}
+
+// TODO: Should be added to reveal/viewer. The input data TransitPath response is in my
+// opinion too coupled with our sdk response, should we split it into two funcitons, one taking in a { x, y, z}[]
+// and another converting the response to said array.
+
+function createWalkablePath(pointData: TransitPathResponse, modelTransformation: SectorModelTransformation) {
```

Comment:
I think the right way to achieve decoupling is to have `getTransitPath` above return something more friendly than 'raw JSON' as above. Also, I like when functions have defined return types as its hard to see what a function returns from the `return`-statements.

Reply:
I have made it so that we have a json to threejs like parser now. I do indeed think that we should have a object that defines the segment, but I am also then thinking we want more in the response object itself before we design this object.